rpcclient/infrastructure.go

```diff
@@ -412,17 +408,29 @@ out:
 			break out
 		default:
 		}
-
-		_, msg, err := c.wsConn.ReadMessage()
-		if err != nil {
-			// Log the error if it's not due to disconnecting.
-			if c.shouldLogReadError(err) {
-				log.Errorf("Websocket receive error from "+
-					"%s: %v", c.config.Host, err)
+		// To check for incoming messages, iterate over the HostAddresses
+		// present with the config. Check if its an active connection
+		// and go ahead with ReadMessage.
+		for i := 0; i < len(c.config.HostAddresses); i++ {
```

Comment:
This doesn't seem like a very efficient way to handle multiple connections.  I would highly suggest just launching a separate `wsInHandler` goroutine per connection that is established instead.  The logic is already based around channels for concurrency.

e.g.
```Go
func (c *Client) wsInHandler(wsConn *websocket.Conn) {
...

		_, msg, err := c.wsConn.ReadMessage()
		if err != nil {
			if c.shouldLogReadError(err) { ... }
			c.balancer.Close(wsConn)
			break out
		}
		c.handleMessage(msg)
...
}
```

Notice that I also changed the `Close` method on the balancer to take the websocket connection which is more robust than dealing with hostnames.  That method would need to be updated accordingly.

Also, I would suggest instead of having `Close` treat an empty string as the disconnect all logic, create a separate method such as `CloseAll` which simply loops the map of open websocket connections and calls `Close` on each one.